tests/scale_test/utils.py

```diff
@@ -0,0 +1,238 @@
+                                    'lt': 9606
+                                }
+                            }
+                        },
+                        {
+                            'range': {
+                                'files.biomaterial_json.biomaterials.content.biomaterial_core.ncbi_taxon_id': {
+                                    'gt': 9606
+                                }
+                            }
+                        }
+                    ],
+                    'should': [
+                        {
+                            'match': {
+                                'files.process_json.processes.content.biomaterial_collection.dissociation_process.dissociation_method': 'FACS'
+                            }
+                        },
+                        {
+                            'match': {
+                                'files.process_json.processes.content.biomaterial_collection.dissociation_process.dissociation_method': 'mouth pipette'
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    }
+    if label:
+        notification['labels'] = label
+    return notification
+
+
+def subscription_probe(lira_url, workflow_name='AdapterSmartSeq2SingleCell'):
+    """A probe to fetch the subscription id.
+
+    This function performs as a probe, fetches the subscription id by hitting the /version endpoint of Lira.
+
+    Args:
+        lira_url (str): A typical Lira url, e.g. https://pipelines.dev.data.humancellatlas.org/
+        workflow_name (str): The workflow to be invoked by the script.
+
+    Returns:
+        str: A valid subscription id.
+    """
+    response = requests.get(harmonize_url(lira_url) + 'version')
+    response.raise_for_status()
+    subscription_id = response.json()['workflow_info'][workflow_name]['subscription_id']
+    return subscription_id
+
+
+def dump_metrics(path, **kwargs):
+    """Dump metrics to a JSON file.
+
+    This function dumps all of the given keyword metrics arguments to a JSON file.
+    Args:
+        path (str): The path to a file.
+        **kwargs: Arbitrary keyword metrics arguments.
+    """
+    metrics = {name: value for name, value in kwargs.items()}
+    with open(path, 'w') as metrics_file:
+        json.dump(metrics, metrics_file)
+
+
+def _prepare_transaction_id():
+    """Transaction UUID generator.
+
+    This function generates a fake valid UUID.
+
+    Returns:
+        str: A fake transaction UUID generated by standard uuid lib.
+    """
+    return str(uuid.uuid4())
+
+
+def harmonize_url(url):
+    """Harmonize the url string.
+
+    This function checks if url ends with slash, if not, add a slash to the url string.
+    Args:
+        url (str): A string of url.
+
+    Returns:
+        str: A string of url ends with slash.
+    """
+    url = str(url)
+    if not url.endswith('/'):
+        url += '/'
+    return url
+
+
+def is_https(url):
```

Comment:
Does this function get used anywhere? If not, can we remove it?

Reply:
Nice catch, I created this function, planned to use it so that this script will only talk to Lira with HTTPS, but ended up with giving up that idea. Will remove it.